smart-campus-service-impl/src/main/java/hu/unideb/smartcampus/service/ejabberd/SharedRosterServiceImpl.java

```diff
@@ -0,0 +1,132 @@
+package hu.unideb.smartcampus.service.ejabberd;
+
+import java.util.List;
+import java.util.Map;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.GenericType;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import hu.unideb.smartcampus.service.ejabberd.request.AddUserRequest;
+import hu.unideb.smartcampus.service.ejabberd.request.CreateGroupRequest;
+import hu.unideb.smartcampus.service.ejabberd.request.DeleteUseRequest;
+import hu.unideb.smartcampus.service.ejabberd.request.GroupRequest;
+import hu.unideb.smartcampus.service.ejabberd.request.InformationRequest;
+
+
+/**
+ * Shared Roster Service Implementation which implements {@link SharedRosterService}.
+ *
+ * <p>
+ * It handles shared roster groups (srg) via REST API which is provided by <b>Ejabberd</b>.
+ * </p>
+ */
+@Service
+public class SharedRosterServiceImpl extends EjabberdService implements SharedRosterService {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SharedRosterServiceImpl.class);
+
+  /**
+   * {@inheritDoc}.
+   */
+  @Override
+  public void addUserToGroup(String user, String group) {
+    LOGGER.info("Add user {} to group {}", user, group);
+    WebTarget client = createClientByUrl(SHARED_ROSTER_GROUP_CREATE_USER_COMMAND);
+    Entity<AddUserRequest> entity =
+        Entity.entity(AddUserRequest.builder().host(DOMAIN).group(group).user(user).build(),
```

Comment:
Response Provider class with generic type (R extends BaseRequest)?

The method can be the following:
<T extends BaseRequest> Response buildResponse(WebTarget webTarget, Entity<T> entity);